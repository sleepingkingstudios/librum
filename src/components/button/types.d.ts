import type { IconDefinition } from '@fortawesome/fontawesome-common-types';

import {
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  ButtonShims,
} from './shims';

import type { ColorTypes } from '../types';

type ButtonContents = {
  children: React.ReactNode,
} | {
  icon?: IconDefinition,
  label: string,
};

type ButtonOptions = {
  className?: string,
  children?: React.ReactNode,
  disabled?: boolean,
  icon?: IconDefinition,
  label?: string,
  outline?: boolean,
  size?: ButtonSize,
  type?: ColorTypes,
} & ButtonContents;

type ButtonSize = ("sm" | "md" | "lg");

type ButtonType = ("button" | "reset" | "submit");

export type LinkButtonProps = {
  url: string,
} & ButtonOptions;

export type PlainButtonProps = {
  htmlType?: ButtonType,
  onClick?: () => void,
} & ButtonOptions;

export type ButtonProps = LinkButtonProps | PlainButtonProps;
