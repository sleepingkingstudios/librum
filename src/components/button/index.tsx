import * as React from 'react';

import type { IconDefinition } from '@fortawesome/fontawesome-common-types';
import { FontAwesomeIcon as Icon } from '@fortawesome/react-fontawesome';
import { Link } from 'react-router-dom';

import { joinClassNames } from '@utils/react-utils';

import type {
  ButtonProps,
  LinkButtonProps,
  PlainButtonProps,
} from './types';

export type { ButtonProps } from './types';

const generateClassName = ({
  className,
  outline = false,
  size,
  type,
}: {
  className?: string,
  outline: boolean,
  size?: string,
  type?: string,
}): string => joinClassNames(
  'btn',
  outline ? 'btn-outline' : null,
  size? `btn-${size}` : null,
  type ? `btn-${type}` : null,
  className,
);

const renderContents = ({
  children,
  icon,
  label,
}: {
  children?: React.ReactNode,
  icon?: IconDefinition,
  label?: string,
}): React.ReactNode => {
  if (children) { return children; }

  if (label) {
    if (!icon) { return label; }

    return (
      <>
        <Icon className="pr-2" icon={icon} />

        { label }
      </>
    )
  }

  return null;
};

const LinkButton = ({
  children,
  className,
  label,
  icon,
  outline,
  size,
  type,
  url,
}: LinkButtonProps): JSX.Element => {
  const joinedClassName = generateClassName({
    className,
    outline,
    size,
    type,
  });

  return (
    <Link className={joinedClassName} to={url}>
      { renderContents({ children, icon, label }) }
    </Link>
  );
};

const PlainButton = ({
  children,
  className,
  disabled = false,
  htmlType = 'button',
  label,
  icon,
  onClick,
  outline,
  size,
  type,
}: PlainButtonProps): JSX.Element => {
  const joinedClassName = generateClassName({
    className,
    outline,
    size,
    type,
  });

  return (
    <button
      className={joinedClassName}
      disabled={disabled}
      onClick={onClick}
      type={htmlType}
    >
      { renderContents({ children, icon, label }) }
    </button>
  );
};

export const Button = (props: ButtonProps): JSX.Element => {
  const { disabled } = props;

  if (('url' in props) && !disabled) { return (<LinkButton {...props} />); }

  return (<PlainButton {...props} />);
};
