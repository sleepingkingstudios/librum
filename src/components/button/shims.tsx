import * as React from 'react';

export const ButtonShims = () => {
  <>
    <button className="btn btn-outline" />
    <button className="btn btn-primary.btn-outline" />
    <button className="btn btn-danger.btn-outline" />
    <button className="btn btn-info.btn-outline" />
    <button className="btn btn-muted.btn-outline" />
    <button className="btn btn-success.btn-outline" />
    <button className="btn btn-warning.btn-outline" />
    <button className="btn btn-sm" />
    <button className="btn btn-md" />
    <button className="btn btn-lg" />
  </>
};
