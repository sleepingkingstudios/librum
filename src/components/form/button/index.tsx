import * as React from 'react';

import { joinClassNames } from '@utils/react-utils';
import type { UpToTwelveColumns } from '@components/types';

interface IFormButtonProps {
  className?: string;
  children: React.ReactNode;
  disabled?: boolean;
  cols?: UpToTwelveColumns;
  type?: "button" | "reset" | "submit";
}

const columnSpanClassName = ({ cols }: { cols: number }): string | null => {
  if (cols === null) { return null; }

  return `col-span-${cols}`;
};

export const FormButton = ({
  className,
  children,
  disabled = false,
  cols = null,
  type = 'button',
}: IFormButtonProps): JSX.Element => {
  const joinedClassName = joinClassNames(
    'form-button',
    columnSpanClassName({ cols }),
    className,
  );

  return (
    <button
      disabled={disabled}
      className={joinedClassName}
      type={type}
    >
      { children }
    </button>
  );
};
