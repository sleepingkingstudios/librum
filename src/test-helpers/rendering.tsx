import * as React from 'react';
import { Provider as ReduxProvider } from 'react-redux';
import {
  MemoryRouter,
  Routes,
  Route
} from "react-router-dom";
import {
  RenderOptions,
  RenderResult,
  render,
} from '@testing-library/react';

import {
  createStore,
  storeType,
} from './store';
import {
  ThemeContext,
  defaultTheme,
} from '@themes/context';
import type { Theme } from '@themes';

interface customRenderOptions extends RenderOptions {
  initialEntries?: Array<string>;
  router?: true | withRouterOptions;
  store?: (storeType | true);
  theme?: (Theme | true);
}

interface withRouterOptions {
  initialEntries?: Array<string>;
  path?: string;
}

export const withRouter = (
  component: React.ReactElement,
  {
    initialEntries = ['/'],
    path,
  }: withRouterOptions,
): React.ReactElement => {
  let wrapped = component;

  if (path) {
    wrapped = (
      <Routes>
        <Route path={path} element={wrapped} />
      </Routes>
    );
  }

  return (
    <MemoryRouter initialEntries={initialEntries}>
      { wrapped }
    </MemoryRouter>
  );
};

export const withStore = (
  component: React.ReactElement,
  { store }: { store: (storeType | true) }
): React.ReactElement => {
  let configuredStore: storeType;

  if (store === true) {
    configuredStore = createStore().store;
  } else {
    configuredStore = store;
  }

  return (
    <ReduxProvider store={configuredStore}>
      { component }
    </ReduxProvider>
  );
};

export const withTheme = (
  component: React.ReactElement,
  { theme }: { theme: (Theme | true) }
): React.ReactElement => {
  let configuredTheme: Theme;

  if (theme === true) {
    configuredTheme = defaultTheme;
  } else {
    configuredTheme = theme;
  }

  return (
    <ThemeContext.Provider value={configuredTheme}>
      { component }
    </ThemeContext.Provider>
  );
};

const customRender = (
  component: React.ReactElement,
  options?: customRenderOptions,
  strictMode = true,
): RenderResult => {
  let wrapped: React.ReactElement = component;

  if (strictMode) {
    wrapped = (<React.StrictMode>{ wrapped }</React.StrictMode>);
  }

  if (!options) { return render(wrapped); }

  if ('router' in options && options.router) {
    if (options.router === true) {
      wrapped = withRouter(wrapped, { initialEntries: options.initialEntries });
    } else {
      wrapped = withRouter(
        wrapped,
        {
          initialEntries: options.initialEntries,
          ...options.router,
        },
      );
    }
  }

  if ('store' in options) {
    wrapped = withStore(wrapped, { store: options.store });
  }

  if ('theme' in options) {
    wrapped = withTheme(wrapped, { theme: options.theme });
  }

  return render(wrapped, options);
};

export { customRender as render };
