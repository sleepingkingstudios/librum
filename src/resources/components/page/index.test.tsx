import * as React from 'react';

import '@testing-library/jest-dom';

import { ResourcePage } from './index';
import { ButtonProps as Button } from '@components/button';
import type { Breadcrumb } from '@components/page';
import { render } from '@test-helpers/rendering';
import type { ResourceApiHooks } from '../../api/types';
import type { ResourcePageOptions } from './types';

jest.mock('@components/page', () => require('@components/page/mocks'));
jest.mock('@components/page/breadcrumbs', () => require('@components/page/breadcrumbs/mocks'));

describe('<ResourcePage />', () => {
  const action = 'published';
  const apiHooks: ResourceApiHooks = {};
  const member = false;
  const resourceName = 'rareBooks';

  it('should display the breadcrumbs', () => {
    const page: ResourcePageOptions = {};
    const { getAllByRole } = render(
      <ResourcePage
        action={action}
        apiHooks={apiHooks}
        member={member}
        page={page}
        resourceName={resourceName}
      />
    );
    const expected = [
      'Home @ /',
      'Rare Books @ /rare-books',
      'Published (active)',
    ];

    const rendered = getAllByRole('listitem');

    expect(rendered).toHaveLength(3);
    expect(rendered.map(item => item.textContent)).toEqual(expected);
  });

  it('should display the heading', () => {
    const page: ResourcePageOptions = {};
    const { getByRole } = render(
      <ResourcePage
        action={action}
        apiHooks={apiHooks}
        member={member}
        page={page}
        resourceName={resourceName}
      />
    );

    const heading = getByRole('heading');

    expect(heading).toBeVisible();
    expect(heading).toHaveTextContent('Published Rare Books');
  });

  it('should match the snapshot', () => {
    const page: ResourcePageOptions = {};
    const { asFragment } = render(
      <ResourcePage
        action={action}
        apiHooks={apiHooks}
        member={member}
        page={page}
        resourceName={resourceName}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  describe('with breadcrumbs: value', () => {
    const page: ResourcePageOptions = {};
    const breadcrumbs: Breadcrumb[] = [
      {
        label: 'Lending Library',
        url: '/lending-library',
      },
    ];

    it('should display the breadcrumbs', () => {
      const { getAllByRole } = render(
        <ResourcePage
          action={action}
          breadcrumbs={breadcrumbs}
          apiHooks={apiHooks}
          member={member}
          page={page}
          resourceName={resourceName}
        />
      );
      const expected = [
        'Lending Library @ /lending-library',
        'Rare Books @ /lending-library/rare-books',
        'Published (active)',
      ];

      const rendered = getAllByRole('listitem');

      expect(rendered).toHaveLength(3);
      expect(rendered.map(item => item.textContent)).toEqual(expected);
    });

    it('should match the snapshot', () => {
      const { asFragment } = render(
        <ResourcePage
          action={action}
          breadcrumbs={breadcrumbs}
          apiHooks={apiHooks}
          member={member}
          page={page}
          resourceName={resourceName}
        />,
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('with contents: component', () => {
    const Contents = ({
      resourceName,
    }: {
      resourceName: string,
    }): JSX.Element => {
      return (<span>Contents for { resourceName }</span>);
    };

    it('should display the contents', () => {
      const page: ResourcePageOptions = { contents: Contents };
      const { getByText } = render(
        <ResourcePage
          action={action}
          apiHooks={apiHooks}
          member={member}
          page={page}
          resourceName={resourceName}
        />
      );

      const rendered = getByText('Contents for rareBooks');

      expect(rendered).toBeVisible();
    });

    it('should match the snapshot', () => {
      const page: ResourcePageOptions = { contents: Contents };
      const { asFragment } = render(
        <ResourcePage
          action={action}
          apiHooks={apiHooks}
          member={member}
          page={page}
          resourceName={resourceName}
        />,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    describe('with afterContents: component', () => {
      const AfterContents = ({
        resourceName,
      }: {
        resourceName: string,
      }): JSX.Element => {
        return (<span>After { resourceName }</span>);
      };
      const page: ResourcePageOptions = {
        afterContents: AfterContents,
        contents: Contents,
      };

      it('should display the contents', () => {
        const { getByText } = render(
          <ResourcePage
            action={action}
            apiHooks={apiHooks}
            member={member}
            page={page}
            resourceName={resourceName}
          />
        );

        expect(getByText('Contents for rareBooks')).toBeVisible();
        expect(getByText('After rareBooks')).toBeVisible();
      });

      it('should match the snapshot', () => {
        const { asFragment } = render(
          <ResourcePage
            action={action}
            apiHooks={apiHooks}
            member={member}
            page={page}
            resourceName={resourceName}
          />,
        );

        expect(asFragment()).toMatchSnapshot();
      });
    });

    describe('with beforeContents: component', () => {
      const BeforeContents = ({
        resourceName,
      }: {
        resourceName: string,
      }): JSX.Element => {
        return (<span>Before { resourceName }</span>);
      };
      const page: ResourcePageOptions = {
        beforeContents: BeforeContents,
        contents: Contents,
      };

      it('should display the contents', () => {
        const { getByText } = render(
          <ResourcePage
            action={action}
            apiHooks={apiHooks}
            member={member}
            page={page}
            resourceName={resourceName}
          />
        );

        expect(getByText('Before rareBooks')).toBeVisible();
        expect(getByText('Contents for rareBooks')).toBeVisible();
      });

      it('should match the snapshot', () => {
        const { asFragment } = render(
          <ResourcePage
            action={action}
            apiHooks={apiHooks}
            member={member}
            page={page}
            resourceName={resourceName}
          />,
        );

        expect(asFragment()).toMatchSnapshot();
      });
    });
  });

  describe('with contents: element', () => {
    const contents = (<span>Custom Contents</span>);

    it('should display the contents', () => {
      const page: ResourcePageOptions = { contents };
      const { getByText } = render(
        <ResourcePage
          action={action}
          apiHooks={apiHooks}
          member={member}
          page={page}
          resourceName={resourceName}
        />
      );

      const rendered = getByText('Custom Contents');

      expect(rendered).toBeVisible();
    });

    it('should match the snapshot', () => {
      const page: ResourcePageOptions = { contents };
      const { asFragment } = render(
        <ResourcePage
          action={action}
          apiHooks={apiHooks}
          member={member}
          page={page}
          resourceName={resourceName}
        />,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    describe('with afterContents: element', () => {
      const afterContents = (<span>After Contents</span>);
      const page: ResourcePageOptions = { afterContents, contents };

      it('should display the contents', () => {
        const { getByText } = render(
          <ResourcePage
            action={action}
            apiHooks={apiHooks}
            member={member}
            page={page}
            resourceName={resourceName}
          />
        );

        expect(getByText('After Contents')).toBeVisible();
        expect(getByText('Custom Contents')).toBeVisible();
      });

      it('should match the snapshot', () => {
        const { asFragment } = render(
          <ResourcePage
            action={action}
            apiHooks={apiHooks}
            member={member}
            page={page}
            resourceName={resourceName}
          />,
        );

        expect(asFragment()).toMatchSnapshot();
      });
    });

    describe('with beforeContents: element', () => {
      const beforeContents = (<span>Before Contents</span>);
      const page: ResourcePageOptions = { beforeContents, contents };

      it('should display the contents', () => {
        const { getByText } = render(
          <ResourcePage
            action={action}
            apiHooks={apiHooks}
            member={member}
            page={page}
            resourceName={resourceName}
          />
        );

        expect(getByText('Before Contents')).toBeVisible();
        expect(getByText('Custom Contents')).toBeVisible();
      });

      it('should match the snapshot', () => {
        const { asFragment } = render(
          <ResourcePage
            action={action}
            apiHooks={apiHooks}
            member={member}
            page={page}
            resourceName={resourceName}
          />,
        );

        expect(asFragment()).toMatchSnapshot();
      });
    });
  });

  describe('with page: { breadcrumbs: element }', () => {
    const breadcrumbs = (<span>Custom Breadcrumbs</span>);
    const page: ResourcePageOptions = { breadcrumbs };

    it('should display the custom breadcrumbs', () => {
      const { getByText } = render(
        <ResourcePage
          action={action}
          apiHooks={apiHooks}
          member={member}
          page={page}
          resourceName={resourceName}
        />
      );

      const rendered = getByText('Custom Breadcrumbs');

      expect(rendered).toBeVisible();
    });

    it('should match the snapshot', () => {
      const { asFragment } = render(
        <ResourcePage
          action={action}
          apiHooks={apiHooks}
          member={member}
          page={page}
          resourceName={resourceName}
        />,
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('with page: { breadcrumbs: value }', () => {
    const breadcrumbs: Breadcrumb[] = [
      {
        label: 'Published Rare Books',
        active: true,
      },
    ];
    const page: ResourcePageOptions = { breadcrumbs };

    it('should display the breadcrumbs', () => {
      const { getAllByRole } = render(
        <ResourcePage
          action={action}
          apiHooks={apiHooks}
          member={member}
          page={page}
          resourceName={resourceName}
        />
      );
      const expected = [
        'Home @ /',
        'Published Rare Books (active)',
      ];

      const rendered = getAllByRole('listitem');

      expect(rendered).toHaveLength(2);
      expect(rendered.map(item => item.textContent)).toEqual(expected);
    });

    it('should match the snapshot', () => {
      const { asFragment } = render(
        <ResourcePage
          action={action}
          apiHooks={apiHooks}
          member={member}
          page={page}
          resourceName={resourceName}
        />,
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('with page: { buttons: value }', () => {
    const buttons: Button[] = [
      {
        label: 'Rock',
        type: 'primary',
      },
      {
        label: 'Paper',
        type: 'success',
      },
      {
        label: 'Scissors',
        type: 'danger',
      },
    ];
    const page: ResourcePageOptions = { buttons };

    it('should display the buttons', () => {
      const { getAllByRole } = render(
        <ResourcePage
          action={action}
          apiHooks={apiHooks}
          member={member}
          page={page}
          resourceName={resourceName}
        />
      );
      const expected = ['Rock', 'Paper', 'Scissors'];

      const rendered = getAllByRole('button');

      expect(rendered).toHaveLength(3);
      expect(rendered.map(button => button.textContent)).toEqual(expected);
    });

    it('should match the snapshot', () => {
      const { asFragment } = render(
        <ResourcePage
          action={action}
          apiHooks={apiHooks}
          member={member}
          page={page}
          resourceName={resourceName}
        />,
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('with page: { heading: value }', () => {
    const page: ResourcePageOptions = { heading: 'Rare Publications' };

    it('should display the heading', () => {
      const { getByRole } = render(
        <ResourcePage
          action={action}
          apiHooks={apiHooks}
          member={member}
          page={page}
          resourceName={resourceName}
        />
      );

      const heading = getByRole('heading');

      expect(heading).toBeVisible();
      expect(heading).toHaveTextContent('Rare Publications');
    });

    it('should match the snapshot', () => {
      const { asFragment } = render(
        <ResourcePage
          action={action}
          apiHooks={apiHooks}
          member={member}
          page={page}
          resourceName={resourceName}
        />,
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('with page: { navigation: element }', () => {
    const navigation = (<span>Custom Navigation</span>);
    const page: ResourcePageOptions = { navigation };

    it('should display the custom navigation', () => {
      const { getByText } = render(
        <ResourcePage
          action={action}
          apiHooks={apiHooks}
          member={member}
          page={page}
          resourceName={resourceName}
        />
      );

      const rendered = getByText('Custom Navigation');

      expect(rendered).toBeVisible();
    });

    it('should match the snapshot', () => {
      const { asFragment } = render(
        <ResourcePage
          action={action}
          apiHooks={apiHooks}
          member={member}
          page={page}
          resourceName={resourceName}
        />,
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('with page: { navigation: value }', () => {
    const navigation = [
      {
        label: 'Nav Item',
        url: '/nav-item',
      },
      {
        label: 'Dropdown',
        items: [
          {
            label: 'Dropdown Item',
            url: '/dropdown-item',
          },
        ],
      },
    ];
    const page = { navigation };

    it('should display the navigation', () => {
      const { getAllByRole } = render(
        <ResourcePage
          action={action}
          apiHooks={apiHooks}
          member={member}
          page={page}
          resourceName={resourceName}
        />
      );
      const expected = [
        'Nav Item @ /nav-item',
        'Dropdown Item @ /dropdown-item',
      ];

      const rendered = getAllByRole('listitem', { name: 'navigation-item' });

      expect(rendered).toHaveLength(2);
      expect(rendered.map(item => item.textContent)).toEqual(expected);
    });

    it('should match the snapshot', () => {
      const { asFragment } = render(
        <ResourcePage
          action={action}
          apiHooks={apiHooks}
          member={member}
          page={page}
          resourceName={resourceName}
        />,
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe('with page: { title: value }', () => {
    const title = 'Rare Books';

    it('should display the title', () => {
      const page = { title };
      const { getByLabelText } = render(
        <ResourcePage
          action={action}
          apiHooks={apiHooks}
          member={member}
          page={page}
          resourceName={resourceName}
        />
      );

      const rendered = getByLabelText('title');

      expect(rendered).toBeVisible();
      expect(rendered).toHaveTextContent('Rare Books');
    });

    it('should match the snapshot', () => {
      const page = { title };
      const { asFragment } = render(
        <ResourcePage
          action={action}
          apiHooks={apiHooks}
          member={member}
          page={page}
          resourceName={resourceName}
        />,
      );

      expect(asFragment()).toMatchSnapshot();
    });

    describe('with page: { subtitle: value }', () => {
      const subtitle = 'the Publishing';
      const page = { subtitle, title };

      it('should display the title and subtitle', () => {
        const { getByLabelText } = render(
          <ResourcePage
            action={action}
            apiHooks={apiHooks}
            member={member}
            page={page}
            resourceName={resourceName}
          />
        );

        const rendered = getByLabelText('title');

        expect(rendered).toBeVisible();
        expect(rendered).toHaveTextContent('Rare Books: the Publishing');
      });

      it('should match the snapshot', () => {
        const { asFragment } = render(
          <ResourcePage
            action={action}
            apiHooks={apiHooks}
            member={member}
            page={page}
            resourceName={resourceName}
          />,
        );

        expect(asFragment()).toMatchSnapshot();
      });
    });
  });

  describe('with scope: value', () => {
    const page: ResourcePageOptions = {};
    const scope = 'lendingLibrary';

    it('should display the breadcrumbs', () => {
      const { getAllByRole } = render(
        <ResourcePage
          action={action}
          apiHooks={apiHooks}
          member={member}
          page={page}
          resourceName={resourceName}
          scope={scope}
        />
      );
      const expected = [
        'Home @ /',
        'Lending Library @ /lending-library',
        'Rare Books @ /lending-library/rare-books',
        'Published (active)',
      ];

      const rendered = getAllByRole('listitem');

      expect(rendered).toHaveLength(4);
      expect(rendered.map(item => item.textContent)).toEqual(expected);
    });

    it('should match the snapshot', () => {
      const { asFragment } = render(
        <ResourcePage
          action={action}
          apiHooks={apiHooks}
          member={member}
          page={page}
          resourceName={resourceName}
          scope={scope}
        />,
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
