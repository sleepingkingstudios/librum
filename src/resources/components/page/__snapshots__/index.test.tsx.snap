// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResourcePage /> should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mock-page"
  >
    <h1>
      Published Rare Books
    </h1>
    <ul
      aria-label="breadcrumbs"
    >
      <li
        aria-label="breadcrumb"
      >
        Home @ /
      </li>
      <li
        aria-label="breadcrumb"
      >
        Rare Books @ /rare-books
      </li>
      <li
        aria-label="breadcrumb"
      >
        Published (active)
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`<ResourcePage /> with breadcrumbs: value should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mock-page"
  >
    <h1>
      Published Rare Books
    </h1>
    <ul
      aria-label="breadcrumbs"
    >
      <li
        aria-label="breadcrumb"
      >
        Lending Library @ /lending-library
      </li>
      <li
        aria-label="breadcrumb"
      >
        Rare Books @ /lending-library/rare-books
      </li>
      <li
        aria-label="breadcrumb"
      >
        Published (active)
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`<ResourcePage /> with contents: component should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mock-page"
  >
    <h1>
      Published Rare Books
    </h1>
    <span>
      Contents for rareBooks
    </span>
    <ul
      aria-label="breadcrumbs"
    >
      <li
        aria-label="breadcrumb"
      >
        Home @ /
      </li>
      <li
        aria-label="breadcrumb"
      >
        Rare Books @ /rare-books
      </li>
      <li
        aria-label="breadcrumb"
      >
        Published (active)
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`<ResourcePage /> with contents: component with afterContents: component should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mock-page"
  >
    <h1>
      Published Rare Books
    </h1>
    <span>
      Contents for rareBooks
    </span>
    <span>
      After rareBooks
    </span>
    <ul
      aria-label="breadcrumbs"
    >
      <li
        aria-label="breadcrumb"
      >
        Home @ /
      </li>
      <li
        aria-label="breadcrumb"
      >
        Rare Books @ /rare-books
      </li>
      <li
        aria-label="breadcrumb"
      >
        Published (active)
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`<ResourcePage /> with contents: component with beforeContents: component should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mock-page"
  >
    <h1>
      Published Rare Books
    </h1>
    <span>
      Before rareBooks
    </span>
    <span>
      Contents for rareBooks
    </span>
    <ul
      aria-label="breadcrumbs"
    >
      <li
        aria-label="breadcrumb"
      >
        Home @ /
      </li>
      <li
        aria-label="breadcrumb"
      >
        Rare Books @ /rare-books
      </li>
      <li
        aria-label="breadcrumb"
      >
        Published (active)
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`<ResourcePage /> with contents: element should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mock-page"
  >
    <h1>
      Published Rare Books
    </h1>
    <span>
      Custom Contents
    </span>
    <ul
      aria-label="breadcrumbs"
    >
      <li
        aria-label="breadcrumb"
      >
        Home @ /
      </li>
      <li
        aria-label="breadcrumb"
      >
        Rare Books @ /rare-books
      </li>
      <li
        aria-label="breadcrumb"
      >
        Published (active)
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`<ResourcePage /> with contents: element with afterContents: element should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mock-page"
  >
    <h1>
      Published Rare Books
    </h1>
    <span>
      Custom Contents
    </span>
    <span>
      After Contents
    </span>
    <ul
      aria-label="breadcrumbs"
    >
      <li
        aria-label="breadcrumb"
      >
        Home @ /
      </li>
      <li
        aria-label="breadcrumb"
      >
        Rare Books @ /rare-books
      </li>
      <li
        aria-label="breadcrumb"
      >
        Published (active)
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`<ResourcePage /> with contents: element with beforeContents: element should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mock-page"
  >
    <h1>
      Published Rare Books
    </h1>
    <span>
      Before Contents
    </span>
    <span>
      Custom Contents
    </span>
    <ul
      aria-label="breadcrumbs"
    >
      <li
        aria-label="breadcrumb"
      >
        Home @ /
      </li>
      <li
        aria-label="breadcrumb"
      >
        Rare Books @ /rare-books
      </li>
      <li
        aria-label="breadcrumb"
      >
        Published (active)
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`<ResourcePage /> with page: { breadcrumbs: element } should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mock-page"
  >
    <h1>
      Published Rare Books
    </h1>
    <span>
      Custom Breadcrumbs
    </span>
  </div>
</DocumentFragment>
`;

exports[`<ResourcePage /> with page: { breadcrumbs: value } should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mock-page"
  >
    <h1>
      Published Rare Books
    </h1>
    <ul
      aria-label="breadcrumbs"
    >
      <li
        aria-label="breadcrumb"
      >
        Home @ /
      </li>
      <li
        aria-label="breadcrumb"
      >
        Published Rare Books (active)
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`<ResourcePage /> with page: { buttons: value } should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mock-page"
  >
    <div
      class="flex flex-col sm:flex-row"
    >
      <h1>
        Published Rare Books
      </h1>
      <ul
        class="justify-end flex flex-col grow sm:flex-row btn-list"
      >
        <button
          class="btn btn-primary"
          type="button"
        >
          Rock
        </button>
        <button
          class="btn btn-success"
          type="button"
        >
          Paper
        </button>
        <button
          class="btn btn-danger"
          type="button"
        >
          Scissors
        </button>
      </ul>
    </div>
    <ul
      aria-label="breadcrumbs"
    >
      <li
        aria-label="breadcrumb"
      >
        Home @ /
      </li>
      <li
        aria-label="breadcrumb"
      >
        Rare Books @ /rare-books
      </li>
      <li
        aria-label="breadcrumb"
      >
        Published (active)
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`<ResourcePage /> with page: { heading: value } should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mock-page"
  >
    <h1>
      Rare Publications
    </h1>
    <ul
      aria-label="breadcrumbs"
    >
      <li
        aria-label="breadcrumb"
      >
        Home @ /
      </li>
      <li
        aria-label="breadcrumb"
      >
        Rare Books @ /rare-books
      </li>
      <li
        aria-label="breadcrumb"
      >
        Published (active)
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`<ResourcePage /> with page: { navigation: element } should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mock-page"
  >
    <span>
      Custom Navigation
    </span>
    <h1>
      Published Rare Books
    </h1>
    <ul
      aria-label="breadcrumbs"
    >
      <li
        aria-label="breadcrumb"
      >
        Home @ /
      </li>
      <li
        aria-label="breadcrumb"
      >
        Rare Books @ /rare-books
      </li>
      <li
        aria-label="breadcrumb"
      >
        Published (active)
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`<ResourcePage /> with page: { navigation: value } should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mock-page"
  >
    <ul
      aria-label="navigation"
    >
      <li
        aria-label="navigation-item"
      >
        Nav Item @ /nav-item
      </li>
      <li>
        Dropdown
        <ul>
          <li
            aria-label="navigation-item"
          >
            Dropdown Item @ /dropdown-item
          </li>
        </ul>
      </li>
    </ul>
    <h1>
      Published Rare Books
    </h1>
    <ul
      aria-label="breadcrumbs"
    >
      <li
        aria-label="breadcrumb"
      >
        Home @ /
      </li>
      <li
        aria-label="breadcrumb"
      >
        Rare Books @ /rare-books
      </li>
      <li
        aria-label="breadcrumb"
      >
        Published (active)
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`<ResourcePage /> with page: { title: value } should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mock-page"
  >
    <span
      aria-label="title"
    >
      Rare Books
    </span>
    <h1>
      Published Rare Books
    </h1>
    <ul
      aria-label="breadcrumbs"
    >
      <li
        aria-label="breadcrumb"
      >
        Home @ /
      </li>
      <li
        aria-label="breadcrumb"
      >
        Rare Books @ /rare-books
      </li>
      <li
        aria-label="breadcrumb"
      >
        Published (active)
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`<ResourcePage /> with page: { title: value } with page: { subtitle: value } should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mock-page"
  >
    <span
      aria-label="title"
    >
      Rare Books: the Publishing
    </span>
    <h1>
      Published Rare Books
    </h1>
    <ul
      aria-label="breadcrumbs"
    >
      <li
        aria-label="breadcrumb"
      >
        Home @ /
      </li>
      <li
        aria-label="breadcrumb"
      >
        Rare Books @ /rare-books
      </li>
      <li
        aria-label="breadcrumb"
      >
        Published (active)
      </li>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`<ResourcePage /> with scope: value should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mock-page"
  >
    <h1>
      Published Rare Books
    </h1>
    <ul
      aria-label="breadcrumbs"
    >
      <li
        aria-label="breadcrumb"
      >
        Home @ /
      </li>
      <li
        aria-label="breadcrumb"
      >
        Lending Library @ /lending-library
      </li>
      <li
        aria-label="breadcrumb"
      >
        Rare Books @ /lending-library/rare-books
      </li>
      <li
        aria-label="breadcrumb"
      >
        Published (active)
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
